networktable.c

```diff
@@ -96,24 +94,23 @@ bool create_network_addresses_table(void)
 // Parse kernel's neighbor cache
 void parse_neighbor_cache(void)
 {
-	FILE* arpfp = NULL;
+	// Open database file
+	if(!dbopen())
+	{
+		logg("parse_arp_cache() - Failed to open DB");
+		return;
+	}
+
 	// Try to access the kernel's neighbor cache
 	// We are only interested in entries which are in either STALE or REACHABLE state
+	FILE* arpfp = NULL;
 	if((arpfp = popen("ip neigh show nud stale nud reachable", "r")) == NULL)
 	{
 		logg("WARN: Command \"ip neigh show nud stale nud reachable\" failed!");
 		logg("      Message: %s", strerror(errno));
 		return;
 	}
 
-	// Open database file
-	if(!dbopen())
-	{
-		logg("parse_arp_cache() - Failed to open DB");
-		pclose(arpfp);
-		return;
-	}
```

Comment:
Why is this moved to before the `ip` command? If the `ip` command fails, the database will remain open.

Reply:
The idea is to not run `ip neigh` when the database is not available. I will add a `dbclose()` in the case the `ip` command fails.